source/grid/grid_generator.cc

```diff
@@ -743,11 +745,16 @@ namespace GridGenerator
     Assert(n_cells_toroidal > 2,
            ExcMessage("Number of cells in toroidal direction has "
                       "to be at least 3."));
+    AssertThrow(phi > 0.0 && phi < 2.0 * numbers::PI + 1.0e-15,
+                ExcMessage("Invalid angle phi specified."));
 
     // the first 8 vertices are in the x-y-plane
-    Point<3> const        p = Point<3>(R, 0.0, 0.0);
-    double const          a = 1. / (1 + std::sqrt(2.0));
-    std::vector<Point<3>> vertices(8 * n_cells_toroidal);
+    Point<3> const p                = Point<3>(R, 0.0, 0.0);
+    double const   a                = 1. / (1 + std::sqrt(2.0));
+    unsigned int   additional_layer = 0; // torus is closed (angle of 2*pi)
```

Comment:
```suggestion
    const unsigned int   additional_layer = phi < 2.0 * numbers::PI - 1.0e-15 ? 1 : 0;
```

Reply:
Couple this with removal of the next two lines.

Reply:
done in https://github.com/dealii/dealii/pull/8916/commits/df2226367fedbca8a7dc1829f1db034db63b7554